### PR TITLE
Fixes #176: Reset errors after submit & new data are entered.

### LINKED ISSUE #176: Error message showing after correcting validation error
### Description

Confusing error message despite updating field.

### Steps to Reproduce

1. Navigate to http://s3.jptdev.com/2016-04-28/string.html
2. Submit form
3. Update age to a number like 132
4. Fails to correct validation error

http://s.natalian.org/2016-04-28/1461831222_1150x1404.png

#### Expected behavior

No error message

#### Actual behavior

Error message doesn't disappear

### Version

https://npmcdn.com/react-jsonschema-form@0.28.0/dist/react-jsonschema-form.js

### PR DESCRIPTION
**Pull request by n1k0**
r=? @leplatrem
/cc @kaihendry

### CODE REVIEW

File: src/components/Form.js [hunk 1]
```diff
@@ -61,8 +61,7 @@ export default class Form extends Component {
 
   onChange = (formData, options={validate: false}) => {
     const liveValidate = this.props.liveValidate || options.validate;
-    const errors = liveValidate ? this.validate(formData) :
-                                  this.state.errors;
+    const errors = liveValidate ? this.validate(formData) : [];
```
Comment (leplatrem) [inline src/components/Form.js]: Will this reset every errors ?

It should reset the error of the field being changed, no ?

Comment (n1k0) [inline src/components/Form.js]: Well that's a possiblily but when `liveValidate` is not enable, you basically have to resubmit the for to have errors back. I could see a UX scenario where you discard errors field by field as soon as they're fixed, but that's the one of the main purpose of `liveValidate` :)

Comment (n1k0) [inline src/components/Form.js]: I'll add a checkbox in the playground to enable/disable live validation so we'll figure out how usable it this.

Comment (n1k0) [inline src/components/Form.js]: Added live validation toggler to the playground, you're definitely right wrt surprising behavior. Il revamp the patch.

Comment (n1k0) [inline src/components/Form.js]: Addressed in https://github.com/mozilla-services/react-jsonschema-form/pull/177/commits/19aa552945c2b5ebf976623fc2eca8c25478e4aa

**Comment by n1k0** [comment]
This will need a bunch of tests added to cover all the corner cases this patch addresses.

**Comment by n1k0** [comment]
Playground has been updated with this patch. Feedback welcome. http://mozilla-services.github.io/react-jsonschema-form/

**Comment by n1k0** [comment]
Well, I think I'm gonna land this. r=me

File: test/Form_test.js [hunk 2]
```diff
@@ -573,6 +573,55 @@ describe("Form", () => {
...
+          Simulate.change(node.querySelectorAll("input[type=text]")[1], {
+            target: {value: "fixed error too"}
+          });
+          submit(node);
+
+          try {
+            Simulate.submit(node);
+          } catch(err) {
+            // Ditto.
+          }
```
Comment (glasserc) [inline test/Form_test.js]: Why this? It seems like the submit just above should take care of this?

Comment (n1k0) [inline test/Form_test.js]: haha good catch, thanks :) this test has been revamped in the meanwhile though.

**Comment by n1k0** [comment]
Fixed in https://github.com/mozilla-services/react-jsonschema-form/commit/23978ea13255c303f710cef1d855660063c99713

File: test/Form_test.js (hunk 2 above)
Comment (n1k0) [inline test/Form_test.js]: Fixed in https://github.com/mozilla-services/react-jsonschema-form/commit/23978ea13255c303f710cef1d855660063c99713

File: src/components/Form.js [hunk 3]
```diff
@@ -60,16 +64,16 @@ export default class Form extends Component {
...
+    const mustValidate = this.props.liveValidate || options.validate;
+    let state;
+    if (!mustValidate) {
+      state = {status: "editing", formData};
+    } else {
+      const errors = this.validate(formData);
+      const errorSchema = toErrorSchema(errors);
+      state = {status: "editing", formData, errors, errorSchema};
+    }
+    setState(this, state, () => {
```
Comment (glasserc) [inline src/components/Form.js]: Could you please elaborate on this change? As far as I can tell it's just a refactor that doesn't change the semantics of the code. Is that right, or does this somehow change error handling?

Comment (n1k0) [inline src/components/Form.js]: No it's a way to perform a single call to setState with the appropriately computed state. Also please comment on the PR global diff view as parts of this commit are now obsolete :/

Comment (n1k0) [inline src/components/Form.js]: Wooops I thought you were commenting on https://github.com/mozilla-services/react-jsonschema-form/pull/163

This PR has now landed, so efforts are more needed on #163. Though if you feel like something is definitely wrong in here, please file an issue about it :)

Comment (glasserc) [inline src/components/Form.js]: No, nothing is wrong per se, I'm just trying to understand the changes you made and how they addressed the issue, so that I can better understand the proect and the context for #163 (which changes some of this code again). Anyhow, I'll look at #163 now..
